### ISSUE TITLE
prompt missing after a recent npm update

### ISSUE THREAD

**Issue by mgol** [alanshaw/david#111]
The prompt is missing after invoking those commands, both in bash & zsh:
```
$ npm -g i david@7.0.2
$ david
```
This is happening on OS X 10.11.5 in Terminal.app.

This is related to the new progress bar in npm 3.10.x. If I revert the internal david npm copy to the latest 3.9, it works fine. With npm 3.10 internally the only way to get the prompt back is to invoke some npm command that shows the new progress bar like `npm install`.

I'd report an issue to npm but npm itself cleans up after its own commands just fine... So I'm not sure if it's an npm issue or is david interacting with it in a weird way.

**Comment by jpaulin** [comment]
My guess: the spinning wheel (progress bar) uses ANSI codes so that depending on which "position" it was left in when exiting from npm, the prompt disappears? Or the same thing happening in David alone. Maybe it is really their interaction. Because interestingly, today I did a small fix into David and then though; "Oh my, now I fixed one thing, but I lose my cursor". After the initial surprise I found two things: first, it doesn't happen always. Second: I found this particular issue. Looking forward into the code, too..

**Comment by jpaulin** [comment]
Can now reproduce the same cursor disappearing behavior, with david 7.0.2, as well as 9.0.0 (latest version at the moment). Noticed that also running a text editor (pico) after having the cursor disappear, restores the cursor back to visible. No wonder, since these text-based software use ANSI and they probably 'restore' some defaults after exit. **And, if all dependencies are up-to-date**, the cursor flaw doesn't happen..

**Comment by jpaulin** [comment]
@mgol By the way, is it that your whole prompt is missing after running 'david', or just the cursor?

**Comment by mgol** [comment]
Just the cursor.

**Comment by mgol** [comment]
In david 10.1.0 I don't see a progress bar and, therefore, the cursor no longer disappears. Was that an intended change or did sth change accidentally?

**Comment by jpaulin** [comment]
Indeed, fixed. I'm on Ubuntu 16.04 and using David from xterm terminal. No longer disappearing cursors. The 9.x.x and 10.1.0 versions have same dependencies, so I would guess it's the new code that fixed this.
